tests/test_init.py

```diff
@@ -1051,3 +1053,60 @@ def test_import_list_formats(data_archive_readonly, cli_runner):
             "MySQL",
             "Shapefile",
         ]
+
+
+@pytest.mark.parametrize(
+    "names,is_okay",
+    [
+        pytest.param(("",), False, id="empty-string"),
+        pytest.param(("a:b",), False, id="special-ascii-chars"),
+        pytest.param(("a\nb",), False, id="ascii-control-chars"),
+        pytest.param(("1a",), False, id="leading-numeral"),
+        pytest.param(("a1",), True, id="trailing-numeral"),
+        pytest.param((".a",), False, id="leading-dot"),
+        pytest.param(("a.",), False, id="trailing-dot"),
+        pytest.param(("a.b",), True, id="contains-dot"),
+        pytest.param(("a ",), False, id="trailing-space"),
+        pytest.param(("a b",), True, id="contains-space"),
+        pytest.param(("COM1",), False, id="windows-reserved-filename"),
+        pytest.param(("com1",), False, id="windows-reserved-filename-ignoring-case"),
+        pytest.param(("ⅶ",), False, id="leading-numeral-unicode"),
+        pytest.param(("Ἢⅶ",), True, id="trailing-numeral-unicode"),
```

Comment:
Why are trailing numbers a problem?

Reply:
they're not, this just tests that that's fine (hence the `True` here)

Reply:
Ah, duh